src/main/java/org/stellar/sdk/TimeBounds.java

```diff
@@ -15,8 +15,8 @@
 	 * @param maxTime 64bit Unix timestamp
 	 */
 	public TimeBounds(long minTime, long maxTime) {
-		if(minTime >= maxTime) {
```

Comment:
I think we should leave `=` here. In 99% cases it's an error. You have almost no cache to predict ledger close time to seconds.

Reply:
I removed the = just to avoid throwing the exception when the time bounds are both unset (when `minTime` and `maxTime` both equal to 0). Perhaps we should change the if condition to
```java
    if (minTime != 0 && maxTime != 0 && minTime >=  maxTime)  {
        throw new IllegalArgumentException("minTime must be < maxTime");
    }
```
The current code also throws the exception when `minTime` is set and `maxTime` is unset. The proposal above allows setting just one of the time bounds. (I am not sure if we want to allow this.)

Reply:
Thanks for mentioning this. Your condition won't throw an exception when both `minTime` and `maxTime` are set and `minTime >= maxTime` We should probably have something like:
```java
// If maxTime set compare to minTime
if (maxTime != 0 && minTime >= maxTime) {
  throw new IllegalArgumentException("minTime must be < maxTime");
}
```
If `maxTime == 0`, any `minTime` value is correct.

Reply:
I agree with your conclusion (yes, we should change the condition as you indicated), but disagree with what you said about my condition. In fact, the proposed code fragment says precisely "throw an exception when both `minTime` and `maxTime` are set and `minTime >= maxTime`"! (I guess you're doing too much multitasking... :-)

Here we have an additional (and unstated) assumption: neither `minTime` nor `maxTime` can be negative. Under this assumption, the condition
```java
(minTime != 0 && maxTime != 0 && minTime >= maxTime)
```
is equivalent to 
```java
(minTime > 0 && maxTime > 0 && minTime >= maxTime)
```
Since the last two clauses above (`maxTime > 0 && minTime >=  maxTime`) imply the first clause (`minTime > 0`), the first clause is redundant and should be dropped. This is the reason your proposal is better.

Perhaps we should also check that `minTime` and `maxTime` are indeed non-negative?
```java
if (minTime < 0 || maxTime < 0) {
    throw new IllegalArgumentException("neither minTime nor maxTime can be negative");
}
else if (maxTime != 0 && minTime >= maxTime) {
    throw new IllegalArgumentException("minTime must be < maxTime");
}
```

Reply:
> I agree with your conclusion (yes, we should change the condition as you indicated), but disagree with what you said about my condition.

Yeah, please ignore my comment, I wrote it after 6am call so I wasn't thinking clearly 😉 . However, we should make sure that we check for the case when `minTime == maxTime` and I agree we should also throw an exception when `minTime < 0 || maxTime < 0`.